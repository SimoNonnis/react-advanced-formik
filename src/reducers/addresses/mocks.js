import { Map, List } from 'immutable';

export const mockedState = Map({
  addresses: Map({
    '5': List([]),
    '35': List([Map({ id: 1 })]),
    '4': List([Map({ id: 1 }), Map({ id: 2 })]),
  }),
  addressesList: [],
  isDisabled: false,
  isLoading: false,
});

export const mockedAddressesList = [
  {
    Id: 100,
    StreetAddress: 'Test street',
    Place: 'anywhere',
  },
];

export const mockedAddressInfo = [
  {
    SubBuilding: 'Test',
    BuildingName: 'Test',
    BuildingNumber: 'Test',
    PrimaryStreet: 'Test',
    PostTown: 'Test',
    County: 'Test',
    Postcode: 'Test',
    Test: 'Test',
  },
  {},
  {},
];

export const mockedAddress = {
  id: 2,
  sub_building: null,
  building_name: null,
  building_number: '892',
  street: 'Alexander Junctions',
  town: 'Graceview',
  county: 'Suffolk',
  postcode: 'WF9 2JY',
  position: 1,
};
