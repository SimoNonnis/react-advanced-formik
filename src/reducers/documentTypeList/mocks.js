export const mockedDocumentTypeList = {
  identification_list_a: [
    {
      list: 'A',
      type: 'proof_of_id_uk_passport',
      label: 'UK Passport',
      description: 'Current valid (signed) full UK Passport',
    },
    {
      list: 'A',
      type: 'proof_of_id_overseas_passport',
      label: 'Overseas Passport',
      description: 'Current valid (signed) overseas Passport',
    },
    {
      list: 'A',
      type: 'proof_of_id_uk_driving_licence',
      label: 'UK Photocard Driving Licence',
      description: 'Current UK photocard Driving Licence (provisional acceptable)',
    },
    {
      list: 'A',
      type: 'proof_of_id_eu_driving_licence',
      label: 'EU Photocard Driving Licence',
      description: 'Current EU photocard Driving Licence',
    },
    {
      list: 'A',
      type: 'proof_of_id_eea_state_id',
      label: 'EEA Member State ID Card',
      description: 'Current valid EEA Member State ID card',
    },
    {
      list: 'A',
      type: 'proof_of_id_biometric_residence_permit',
      label: 'Biometric Residence Permit',
      description: 'Current biometric residence permit issued by UK Border Agency',
    },
    {
      list: 'A',
      type: 'proof_of_id_firearm_shotgun_certificate',
      label: 'Current Firearms Certificate',
      description: 'Current Firearms/Shotgun Certificate',
    },
    {
      list: 'A',
      type: 'proof_of_id_uk_military_card',
      label: 'UK Military Identity Card',
      description: 'Current UK Military Identity Card',
    },
  ],
  identification_list_b: [
    {
      list: 'B',
      type: 'proof_of_id_bank_society_or_union_statement',
      label: 'Bank Statement',
      description:
        'Bank, Building Society or Credit Union Statement dated within the last 3 months (not printed off the internet)',
    },
    {
      list: 'B',
      type: 'proof_of_id_benefits_or_pensions_notification_letter',
      label: 'Benefits or Pensions Letter',
      description: 'Benefits or pensions notification letter confirming the right to benefit',
    },
    {
      list: 'B',
      type: 'proof_of_id_blue_disabled_drivers_pass',
      label: 'Blue Disabled Drivers Pass',
      description: 'Blue disabled drivers pass',
    },
    {
      list: 'B',
      type: 'proof_of_id_current_uk_paper_driving_licence',
      label: 'UK Paper Driving Licence',
      description: 'Current UK paper driving licence',
    },
    {
      list: 'B',
      type: 'proof_of_id_local_authority_tax_bill_or_council_tax_bill',
      label: 'Tax Bill',
      description:
        'Local authority tax bill/council tax bill (we can only accept bills dated until the end of June of the year the client contract is signed)',
    },
    {
      list: 'B',
      type: 'proof_of_id_medical_card_or_certificate',
      label: 'Medical Card/Certificate',
      description: 'Medical Card/Certificate',
    },
    {
      list: 'B',
      type: 'proof_of_id_national_insurance_card',
      label: 'National Insurance Card',
      description: 'National Insurance Card',
    },
    {
      list: 'B',
      type: 'proof_of_id_uk_birth_certificate',
      label: 'UK Birth Certificate',
      description: 'UK Birth Certificate',
    },
    {
      list: 'B',
      type: 'proof_of_id_utility_bill_utility_statement_or_certificate_letter',
      label: 'Utility Bill',
      description:
        'Utility Bill/Utility Statement or Certificate/Letter from a supplier of utilities dated within the last 3 months',
    },
  ],
  proof_of_address_list_c: [
    {
      list: 'C',
      type: 'proof_of_address_bank_statement',
      label: 'Bank Statement',
      description: 'Bank Statement (dated in the last three months) – may be an e-copy',
    },
    {
      list: 'C',
      type: 'proof_of_address_credit_card_statement',
      label: 'Credit Card Statement',
      description: 'Credit Card Statement (dated in the last three months) – may be an e-copy',
    },
    {
      list: 'C',
      type: 'proof_of_address_council_tax_bill',
      label: 'Council Tax Bill',
      description:
        'Council Tax bill (we can only accept bills dated until the end of June of the year the client contract is signed)',
    },
    {
      list: 'C',
      type: 'proof_of_address_current_mortgage_statement',
      label: 'Mortgage Statement',
      description:
        'Current mortgage statement (correspondence address and address the mortgage applies to must be the same)',
    },
    {
      list: 'C',
      type: 'proof_of_address_current_tv_licence',
      label: 'TV Licence',
      description: 'Current TV licence',
    },
    {
      list: 'C',
      type: 'proof_of_address_driving_licence_with_current_address',
      label: 'Driving Licence',
      description:
        'Driving licence showing current address (paper OR card version) (provisional acceptable) (not acceptable if used as ID)',
    },
    {
      list: 'C',
      type: 'proof_of_address_current_home_insurance_policy',
      label: 'Home Insurance Policy',
      description: 'Homeowner’s current home insurance policy schedule',
    },
    {
      list: 'C',
      type: 'proof_of_address_home_service_provider_bill',
      label: 'Broadband or Digital TV Bill',
      description:
        'Home service provider bill, such as broadband or digital TV dated within the last three months',
    },
    {
      list: 'C',
      type: 'proof_of_address_letter_from_the_employer_on_company',
      label: 'Letter from the Employer',
      description:
        'Letter from the employer on company headed paper, signed and dated within the last three months (an email from a verified company email address is acceptable)',
    },
    {
      list: 'C',
      type: 'proof_of_address_house_benefits_documentation',
      label: 'Housing Allowance/Benefit',
      description:
        'Recent documentation confirming the applicant will be/is receiving local housing allowance or housing benefit',
    },
    {
      list: 'C',
      type: 'proof_of_address_tenancy_agreement',
      label: 'Tenancy Agreement',
      description:
        'Tenancy agreement signed and dated within the last six months (the customer being checked must be a named tenant on the tenancy agreement)',
    },
    {
      list: 'C',
      type: 'proof_of_address_utility_bill',
      label: 'Utility Bill',
      description: 'Utility bill dated within the last three months',
    },
    {
      list: 'C',
      type: 'proof_of_address_letter_from_the_nhs',
      label: 'Letter from the NHS',
      description: 'Letter from the NHS writing to confirm the customer is living at the address',
    },
    {
      list: 'C',
      type: 'proof_of_address_letter_from_bank',
      label: 'Letter from the Bank',
      description:
        'Letter from bank to confirm the customer is living at the property – no other bank letter is acceptable',
    },
  ],
};
