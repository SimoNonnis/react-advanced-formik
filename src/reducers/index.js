import { combineReducers } from 'redux-immutable';
import { reducers as defaultReducers } from '@housesimple/redux';

import documentTypeListReducer from 'reducers/documentTypeList';

import vendorsReducer from 'reducers/vendors';
import updateVendorReducer from 'reducers/updateVendor';
import addVendorReducer from 'reducers/addVendor';
import deleteVendorReducer from 'reducers/deleteVendor';

import addressesReducer from 'reducers/addresses';

import proofOfIdReducer from 'reducers/proofOfId';
import proofOfAddressReducer from 'reducers/proofOfAddress';

import banner from 'reducers/banner';

export default combineReducers(
  Object.assign({}, defaultReducers, {
    documentTypeList: documentTypeListReducer,
    vendors: vendorsReducer,
    updateVendor: updateVendorReducer,
    addVendor: addVendorReducer,
    deleteVendor: deleteVendorReducer,
    banner,
    addresses: addressesReducer,
    proofOfId: proofOfIdReducer,
    proofOfAddress: proofOfAddressReducer,
  })
);
