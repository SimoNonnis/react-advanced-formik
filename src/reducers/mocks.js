export const mockedVendors = [
  {
    id: 4,
    primary: true,
    type: 'legal_owner',
    present_at_home_visit: true,
    title: 'Mr',
    first_names: 'Finley',
    last_name: 'Powell',
    email_address: 'lucas.graham@example.net',
    phone_number: '07000000000',
    date_of_birth: '1974-12-01',
    occupation: 'developer',
    politically_exposed: true,
    living_at_the_property_you_are_selling: false,
    living_at_the_property_you_are_selling_for_at_least_twelve_months: null,
    country_of_residence: 'United Kingdom',
    addresses: [
      {
        id: 1,
        sub_building: null,
        building_name: null,
        building_number: '202',
        street: 'Amelia Courts',
        town: 'Lake Staceyview',
        county: 'Suffolk',
        postcode: 'DD1 2NF',
        position: 0,
      },
      {
        id: 2,
        sub_building: null,
        building_name: null,
        building_number: '86',
        street: 'Evelyn Springs',
        town: 'Walkermouth',
        county: 'Suffolk',
        postcode: 'SG6 1PR',
        position: 1,
      },
    ],
    owned_since_month: 'January',
    owned_since_year: '2000',
    reason_for_selling: 'Selling to buy a larger house',
    proof_of_address_files: [
      {
        id: 2,
        list: 'C',
        type: 'proof_of_address_bank_statement',
        label: 'Bank Statement',
        mime_type: 'image/jpeg',
        path: '/admin/check/5/fetch-file/2',
      },
    ],
    identification_files: [
      {
        id: 8,
        list: 'B',
        type: 'proof_of_id_bank_society_or_union_statement',
        label: 'Bank Statement',
        mime_type: 'image/png',
        path: '/admin/check/7/fetch-file/8',
      },
      {
        id: 9,
        list: 'B',
        type: 'proof_of_id_blue_disabled_drivers_pass',
        label: 'Blue Disabled Drivers Pass',
        mime_type: 'image/jpeg',
        path: '/admin/check/7/fetch-file/9',
      },
    ],
  },
  {
    id: 34,
    primary: false,
    type: 'legal_owner',
    present_at_home_visit: false,
    title: 'Mr',
    first_names: 'Ray',
    last_name: 'Ross',
    email_address: 'danielle92@example.net',
    phone_number: '07000000000',
    date_of_birth: '1996-01-24',
    occupation: 'developer',
    politically_exposed: false,
    living_at_the_property_you_are_selling: false,
    living_at_the_property_you_are_selling_for_at_least_twelve_months: null,
    country_of_residence: 'France',
    addresses: [
      {
        id: 5,
        sub_building: null,
        building_name: null,
        building_number: '92',
        street: 'Dave Drive',
        town: 'Bellview',
        county: 'Suffolk',
        postcode: 'G65 9BG',
        position: 0,
      },
    ],
    owned_since_month: 'January',
    owned_since_year: '2000',
    reason_for_selling: 'Selling to buy a larger house',
    proof_of_address_files: [
      {
        id: 5,
        list: 'C',
        type: 'proof_of_address_current_tv_licence',
        label: 'TV Licence',
        mime_type: 'image/jpeg',
        path: '/admin/check/6/fetch-file/5',
      },
    ],
    identification_files: [
      {
        id: 3,
        list: 'B',
        type: 'proof_of_id_uk_birth_certificate',
        label: 'UK Birth Certificate',
        mime_type: 'image/jpeg',
        path: '/admin/check/8/fetch-file/3',
      },
      {
        id: 4,
        list: 'B',
        type: 'proof_of_id_utility_bill_utility_statement_or_certificate_letter',
        label: 'Utility Bill',
        mime_type: 'image/jpeg',
        path: '/admin/check/8/fetch-file/4',
      },
    ],
  },
];

export const mockedVendor = {
  id: 55,
  primary: false,
  type: 'legal_owner',
  present_at_home_visit: true,
  title: 'mr',
  first_names: 'Jack',
  last_name: 'London',
  email_address: null,
  phone_number: null,
  date_of_birth: null,
  occupation: null,
  politically_exposed: null,
  living_at_the_property_you_are_selling: true,
  living_at_the_property_you_are_selling_for_at_least_twelve_months: true,
  country_of_residence: 'United Kingdom',
  addresses: [
    {
      id: 143,
      sub_building: null,
      building_name: null,
      building_number: '4',
      street: 'Victoria Tunnel',
      town: 'Port Louisberg',
      county: 'Suffolk',
      postcode: 'RG14 2EL',
      position: 0,
    },
  ],
  owned_since_month: null,
  owned_since_year: null,
  reason_for_selling: null,
  proof_of_address_files: [
    {
      id: 11,
      list: 'C',
      type: 'proof_of_address_current_mortgage_statement',
      label: 'Mortgage Statement',
      mime_type: 'image/png',
      path: '/admin/check/10/fetch-file/11',
    },
  ],
  identification_files: [
    {
      id: 6,
      list: 'A',
      type: 'proof_of_id_overseas_passport',
      label: 'Overseas Passport',
      mime_type: 'image/png',
      path: '/admin/check/9/fetch-file/6',
    },
  ],
};
