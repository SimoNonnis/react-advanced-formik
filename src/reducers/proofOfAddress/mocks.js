import { Map, List } from 'immutable';

export const mockedState = Map({
  vendorsProofOfAddress: Map({
    '4': List([
      Map({
        id: 2,
        list: 'C',
        type: 'proof_of_address_bank_statement',
        label: 'Bank Statement',
        mime_type: 'image/jpeg',
        path: '/admin/check/5/fetch-file/2',
      }),
    ]),
    '34': List([]),
  }),
  isLoading: Map({
    '4': false,
    '34': false,
  }),
});

export const mockedPoAd = {
  id: 40,
  list: 'C',
  type: 'proof_of_address_current_mortgage_statement',
  label: 'Mortgage Statement',
  mime_type: 'image/jpeg',
  path: '/admin/check/5/fetch-file/2',
};
