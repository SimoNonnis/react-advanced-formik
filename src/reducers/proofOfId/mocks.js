import { Map, List } from 'immutable';

export const mockedState = Map({
  vendorsProofOfId: Map({
    '4': List([
      Map({
        id: 8,
        list: 'B',
        type: 'proof_of_id_bank_society_or_union_statement',
        label: 'Bank Statement',
        mime_type: 'image/png',
        path: '/admin/check/7/fetch-file/8',
      }),
    ]),
    '34': List([]),
  }),
  isLoading: Map({
    '4': false,
    '34': false,
  }),
});

export const mockedPoId = {
  id: 80,
  list: 'A',
  type: 'proof_of_id_overseas_passport',
  label: 'Overseas Passport',
  mime_type: 'image/png',
  path: '/admin/check/7/fetch-file/8',
};
