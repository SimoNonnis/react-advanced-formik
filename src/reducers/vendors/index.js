import { handleActions } from 'redux-actions';
import { Map, List } from 'immutable';

import {
  requestGetVendors,
  successGetVendors,
  rejectGetVendors,
  successAddVendor,
  successDeleteVendor,
} from 'actions/vendors';

const initialState = new Map({
  isFetchingVendors: false,
  vendors: List(),
});

const getVendorIndex = (state, id) => state.get('vendors').findIndex(vendor => id === vendor.id);

const vendorsReducer = handleActions(
  {
    [requestGetVendors]: state => state.set('isFetchingVendors', true),
    [successGetVendors]: (state, { payload }) =>
      state.merge(
        Map({
          isFetchingVendors: false,
          vendors: List(payload),
        })
      ),
    [successAddVendor]: (state, { payload }) => {
      const updatedVendorsList = state.get('vendors').push(payload);

      return state.set('vendors', updatedVendorsList);
    },
    [successDeleteVendor]: (state, { payload: vendorId }) => {
      const updatedVendorsList = state.get('vendors').delete(getVendorIndex(state, vendorId));

      return state.set('vendors', updatedVendorsList);
    },
    [rejectGetVendors]: () => initialState,
  },
  initialState
);

export default vendorsReducer;
export { initialState };
