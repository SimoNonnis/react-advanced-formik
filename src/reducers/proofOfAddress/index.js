import { handleActions } from 'redux-actions';
import { Map, List, fromJS } from 'immutable';

import { successGetVendors, successAddVendor } from 'actions/vendors';
import {
  requestUploadPoAd,
  successUploadPoAd,
  rejectUploadPoAd,
  requestDeletePoAd,
  successDeletePoAd,
  rejectDeletePoAd,
} from 'actions/documents';

const initialState = new Map({
  isLoading: Map(),
  vendorsProofOfAddress: Map(),
});

const getPoAdIndex = (state, vendorId, documentId) =>
  state.getIn(['vendorsProofOfAddress', vendorId]).findIndex(PoAd => documentId === PoAd.get('id'));

const proofOfAddressReducer = handleActions(
  {
    [successGetVendors]: (state, { payload }) => {
      const proofOfAddressFiles = {};
      const isLoading = {};

      function setVendorIdKey(vendor) {
        proofOfAddressFiles[vendor.id] = [...vendor.proof_of_address_files];
      }

      function setIsLoadingKey(vendor) {
        isLoading[vendor.id] = false;
      }

      payload.map(vendor => setIsLoadingKey(vendor));
      payload.map(vendor => setVendorIdKey(vendor));

      return state
        .set('vendorsProofOfAddress', Map(fromJS(proofOfAddressFiles)))
        .set('isLoading', Map(fromJS(isLoading)));
    },
    [successAddVendor]: (state, { payload: { id } }) => {
      const vendorId = id.toString();

      return state
        .setIn(['isLoading', vendorId], false)
        .setIn(['vendorsProofOfAddress', vendorId], List());
    },
    [requestUploadPoAd]: (state, { payload: { vendorId } }) =>
      state.setIn(['isLoading', vendorId], true),
    [successUploadPoAd]: (state, { payload: { vendorId, data } }) =>
      state
        .setIn(['vendorsProofOfAddress', vendorId, -1], Map(data))
        .setIn(['isLoading', vendorId], false),
    [rejectUploadPoAd]: (state, { payload: { vendorId } }) =>
      state.setIn(['isLoading', vendorId], false),
    [requestDeletePoAd]: (state, { payload: { vendorId } }) =>
      state.setIn(['isLoading', vendorId], true),
    [successDeletePoAd]: (state, { payload: { vendorId, documentId } }) => {
      const vendorPoAd = state.getIn(['vendorsProofOfAddress', vendorId]);
      const newPoAdArray = vendorPoAd.delete(getPoAdIndex(state, vendorId, documentId));

      return state
        .setIn(['vendorsProofOfAddress', vendorId], newPoAdArray)
        .setIn(['isLoading', vendorId], false);
    },
    [rejectDeletePoAd]: (state, { payload: { vendorId } }) =>
      state.setIn(['isLoading', vendorId], false),
  },
  initialState
);

export default proofOfAddressReducer;
export { initialState };
