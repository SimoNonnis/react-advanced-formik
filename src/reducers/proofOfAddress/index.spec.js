import { successGetVendors, successAddVendor } from 'actions/vendors';

import {
  requestUploadPoAd,
  successUploadPoAd,
  rejectUploadPoAd,
  requestDeletePoAd,
  successDeletePoAd,
  rejectDeletePoAd,
} from 'actions/documents';

import { mockedVendors } from 'reducers/mocks';
import { mockedState, mockedPoAd } from './mocks';

import proofOfAddressReducer, { initialState } from './';

describe('Test proofOfAddressReducer', () => {
  it('should return initial state', () => {
    expect(proofOfAddressReducer(undefined, {})).toBe(initialState);
  });

  it('should build isLoading obj and vendorsProofOfAddress obj by vendor ID', () => {
    const reducer = proofOfAddressReducer(initialState, successGetVendors(mockedVendors));
    expect(reducer).toMatchSnapshot();
  });
  it('should set new POAD empty array and isLoading false by vendor ID', () => {
    const reducer = proofOfAddressReducer(initialState, successAddVendor({ id: 3 }));
    expect(reducer).toMatchSnapshot();
  });

  it('should set isLoading to true after calling requestUploadPoAd()', () => {
    const reducer = proofOfAddressReducer(mockedState, requestUploadPoAd({ vendorId: '4' }));
    expect(reducer.getIn(['isLoading', '4'])).toBe(true);
  });
  it('should add new file obj to vendorsProofOfAddress', () => {
    const reducer = proofOfAddressReducer(
      mockedState,
      successUploadPoAd({ vendorId: '34', data: mockedPoAd })
    );
    expect(reducer).toMatchSnapshot();
  });
  it('should set isLoading to false after calling rejectUploadPoAd()', () => {
    const reducer = proofOfAddressReducer(mockedState, rejectUploadPoAd({ vendorId: '4' }));
    expect(reducer.getIn(['isLoading', '4'])).toBe(false);
  });

  it('should set isLoading to true after calling requestDeletePoAd()', () => {
    const reducer = proofOfAddressReducer(mockedState, requestDeletePoAd({ vendorId: '4' }));
    expect(reducer.getIn(['isLoading', '4'])).toBe(true);
  });
  it('should delete file obj from vendorsProofOfAddress', () => {
    const reducer = proofOfAddressReducer(
      mockedState,
      successDeletePoAd({ vendorId: '4', documentId: '1' })
    );
    expect(reducer).toMatchSnapshot();
  });
  it('should set isLoading to false after calling rejectDeletePoAd()', () => {
    const reducer = proofOfAddressReducer(mockedState, rejectDeletePoAd({ vendorId: '4' }));
    expect(reducer.getIn(['isLoading', '4'])).toBe(false);
  });
});
