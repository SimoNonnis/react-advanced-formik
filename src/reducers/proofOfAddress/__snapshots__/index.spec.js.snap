// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test proofOfAddressReducer should add new file obj to vendorsProofOfAddress 1`] = `
Object {
  "isLoading": Object {
    "34": false,
    "4": false,
  },
  "vendorsProofOfAddress": Object {
    "34": Array [
      Object {
        "id": 40,
        "label": "Mortgage Statement",
        "list": "C",
        "mime_type": "image/jpeg",
        "path": "/admin/check/5/fetch-file/2",
        "type": "proof_of_address_current_mortgage_statement",
      },
    ],
    "4": Array [
      Object {
        "id": 2,
        "label": "Bank Statement",
        "list": "C",
        "mime_type": "image/jpeg",
        "path": "/admin/check/5/fetch-file/2",
        "type": "proof_of_address_bank_statement",
      },
    ],
  },
}
`;

exports[`Test proofOfAddressReducer should build isLoading obj and vendorsProofOfAddress obj by vendor ID 1`] = `
Object {
  "isLoading": Object {
    "34": false,
    "4": false,
  },
  "vendorsProofOfAddress": Object {
    "34": Array [
      Object {
        "id": 5,
        "label": "TV Licence",
        "list": "C",
        "mime_type": "image/jpeg",
        "path": "/admin/check/6/fetch-file/5",
        "type": "proof_of_address_current_tv_licence",
      },
    ],
    "4": Array [
      Object {
        "id": 2,
        "label": "Bank Statement",
        "list": "C",
        "mime_type": "image/jpeg",
        "path": "/admin/check/5/fetch-file/2",
        "type": "proof_of_address_bank_statement",
      },
    ],
  },
}
`;

exports[`Test proofOfAddressReducer should delete file obj from vendorsProofOfAddress 1`] = `
Object {
  "isLoading": Object {
    "34": false,
    "4": false,
  },
  "vendorsProofOfAddress": Object {
    "34": Array [],
    "4": Array [],
  },
}
`;

exports[`Test proofOfAddressReducer should set new POAD empty array and isLoading false by vendor ID 1`] = `
Object {
  "isLoading": Object {
    "3": false,
  },
  "vendorsProofOfAddress": Object {
    "3": Array [],
  },
}
`;
